scripts/solana_read_fcd_onchain.ts

```diff
@@ -0,0 +1,47 @@
+import * as anchor from '@project-serum/anchor';
+import {Program} from '@project-serum/anchor';
+import {PublicKey, Connection} from '@solana/web3.js';
+import fs from 'fs';
+
+import {derivePDAFromFCDKey, getReturnLog} from './solana-utils';
+
+/*
+ Execution example:
+  SOLANA_CHAIN_PROGRAM_ID=4SPgs3L7Ey9VyRuZwx4X3y86LSAZXP2Hhpz9Sps4v3iT
+  SOLANA_CALLER_PROGRAM_ID=BAxZZXMFKHn4cw2NJTM83uLSuuhF3BBq76WLKvDkz5xj
+  ANCHOR_PROVIDER_URL="https://api.devnet.solana.com"
+  ANCHOR_WALLET=~/.config/solana/id.json
+  ts-node ./scripts/solana_read_fcd_onchain.ts
+*/
+
+const chainProgramId = process.env.SOLANA_CHAIN_PROGRAM_ID as string;
+const callerProgramId = process.env.SOLANA_CALLER_PROGRAM_ID as string;
+
+const IDL = JSON.parse(fs.readFileSync('./solana-idl/caller.json', 'utf8'));
+
+const main = async () => {
+  const provider = anchor.AnchorProvider.env();
+  anchor.setProvider(provider);
+
+  const callerProgram = new Program(IDL, new PublicKey(callerProgramId), provider);
+
+  const key = 'BTC-USD';
```

Comment:
I didn't get what is this key, is it always this value? maybe a setting fits better.

Reply:
In this case is an example, it should work for any FCD we create on Solana